Amethyst/AppDelegate.swift

```diff
@@ -173,6 +184,66 @@ class AppDelegate: NSObject, NSApplicationDelegate {
             }
         }
     }
+
+    private func populateLayoutsMenu() {
+        guard let layoutsMenuItem = layoutsMenuItem,
+              let submenu = layoutsMenuItem.submenu else {
+            return
+        }
+
+        // Clear existing items
+        submenu.removeAllItems()
+
+        // Get enabled layout keys from user configuration
+        let enabledLayoutKeys = UserConfiguration.shared.layoutKeys()
+
+        // Get all available layouts with their display names
+        let availableLayouts = LayoutType<SIApplication.Window>.availableLayoutStrings()
+
+        // Get current layout
+        let focusedScreenManager = windowManager?.focusedScreenManager()
+        var currentLayoutKey = focusedScreenManager?.currentLayout?.layoutKey
+
+        // If no focused screen manager, fallback to the first screen manager
+        if focusedScreenManager == nil, let firstScreenManager = windowManager?.screenManager(at: 0) {
+            currentLayoutKey = firstScreenManager.currentLayout?.layoutKey
+        }
+
+        // Filter to only enabled layouts and add menu items
+        for layoutKey in enabledLayoutKeys {
+            guard let layoutInfo = availableLayouts.first(where: { $0.key == layoutKey }) else {
+                continue
+            }
+
+            let menuItem = NSMenuItem(title: layoutInfo.name, action: #selector(selectLayout(_:)), keyEquivalent: "")
+            menuItem.target = self
+            menuItem.representedObject = layoutKey
+
+            // Mark current layout with checkmark
+            let isCurrentLayout = layoutKey == currentLayoutKey
+            menuItem.state = isCurrentLayout ? .on : .off
+
+            submenu.addItem(menuItem)
+        }
+
+        // If no layouts are enabled, show a disabled message
+        if enabledLayoutKeys.isEmpty {
+            let noLayoutsItem = NSMenuItem(title: "No layouts enabled", action: nil, keyEquivalent: "")
+            noLayoutsItem.isEnabled = false
+            submenu.addItem(noLayoutsItem)
+        }
```

Comment:
This is minor, but can you just do this check before the loop and then return if it's empty?

Reply:
Fixed. Moved the empty layouts check to the beginning of `populateLayoutsMenu()` with an early return.